lib/sales_engine.rb

```diff
@@ -99,11 +99,55 @@ def customer_merchants(id)
   end
 
   def total(id)
+  return 0 if !invoices.find_by_id(id).is_paid_in_full?
   selected_invoices = invoice_items.find_all_by_invoice_id(id)
   totals_of_invoice_items = selected_invoices.map do |invoice_item|
     (invoice_item.quantity * invoice_item.unit_price)
     end
     totals_of_invoice_items.reduce(:+)
   end
```

Comment:
@erikaannesmith @Tyjoo26 
Whenever you see a map followed by a reduce, there is likely the opportunity to refactor to using a single reduce to get the job done.